src/wrapper.hpp

```diff
@@ -28,6 +28,8 @@ extern "C" {
 
 namespace adblock {
 
+bool setDomainResolver(void (*resolver)(const char* host, uintptr_t* start, uintptr_t* end));
```

Comment:
is a function pointer the only way to do this in Rust? I think ideally we would move ffi into brave-core and pass in a `base::RepeatingCallback`
In this case it seems fine because we're calling into a static method, but it would be nice to have an example that is safer for other types of callbacks

Reply:
I believe the only way to cross this behavior across the C/Rust FFI is to use a function pointer.

I would be happy to "wrap" the function pointer in something a little more ergonomic on the C++ side here, e.g. something that converted the `const char*` argument into a `std::string&`. I did try using a C++11 lambda for this purpose but I ran into issues coercing it into a function pointer of the corresponding signature.